components/Compare/Compare.js

```diff
@@ -0,0 +1,187 @@
+import StyledList from "../CarsList/StyledList";
+import {
+  StyledImage,
+  StyledLikeButton,
+  StyledResultDiv,
+} from "../Favorites/StyledFavorites";
+import Image from "next/image";
+import styled from "styled-components";
+import { StyledSubmitButton } from "../FilterForm/StyledForm";
+
+const StyledCompareList = styled.ul`
+  list-style: none;
+  padding: 0px;
+  margin-top: 5px;
+  display: grid;
+  gap: 10px;
+  justify-items: center;
+  grid-template-columns: 1fr 1fr;
+  background-color: grey;
+  font-size: 1.2em;
+`;
+
+const StyledNameCompareH3 = styled.h3`
+  margin: 0;
+`;
+
+function Compare({
+  cars,
+  onToggleFavorite,
+  onToggleCompared,
+  clearComparedList,
+}) {
+  return (
+    <>
+      {cars.filter((car) => car.isCompared).length < 1 ? (
```

Comment:
Instead of running `cars.filter(...)` for every entry, you can do it once at the beginning like:

```js
const carsToCompare = cars.filter((car) => car.isCompared);

return (
  <>
    {
      !carsToCompare.length
        ? "No cars to compare"
        : carsToCompare.map(...)
      }
  </>
)
```